docs/build.py

```diff
@@ -141,7 +141,9 @@ def cmd(s):
 add_translation.exec(temp_directory)
 
 print("Running doxygen")
-cmd('cd "{temp_directory}" && doxygen Doxyfile'.format(temp_directory=temp_directory))
+os.chdir(temp_directory)
+cmd('doxygen Doxyfile')
+os.chdir(base_path)
```

Comment:
The only issue I see here is assumptions are being made. You are assuming that the user is running the script from the same folder that the script is located in...

It would be better to modify the `cmd` function to take an additional parameter of `cwd`...

```python

def cmd(s, cwd=None):
    if cwd is None:
        cwd = os.getcwd()

    cur_dir = os.getcwd()
    os.chdir(cwd)
    print("")
    print(s)
    print("-------------------------------------")
    r = os.system(s)
    os.chdir(cur_dir)
    if r != 0:
        print("Exit build due to previous error")
        exit(-1)


print("Running doxygen")
cmd('doxygen Doxyfile', temp_directory)
```

I personally feel this solution is a better one as it doesn't make any assumptions as to what the cwd is.

Reply:
> ```python
> doc_builder.EMIT_WARNINGS = False
> ```

Tested and indeed it suppresses a very large number of "missing documentation" warnings.



> I personally feel this solution is a better one as it doesn't make any assumptions as to what the cwd is.

Clearly this is an enhancement.  I added it only using different variable (new) names for code clarity.

Reply:
yeah that's fine. The suggestion was made because honestly I am not sure if there are any other things that using that `cmd` function that would need to have their working directory supplied..... You may want to take a breeze through mthe file and see if that function is called anywhere else that has the directory being changed like what was being done to compile doxygen.

what you didn't change was the lines where you call `cmd` for the doxygen build.

Reply:
> what you didn't change was the lines where you call `cmd` for the doxygen build.

Oh!  Good idea!

Reply:
Done!